lib/auth.js

```diff
@@ -9,6 +9,11 @@ function getSocketSignature(token, channel, socketID, data) {
   }
 
   result.auth = token.key + ':' + token.sign(signatureData.join(":"));
+
+  if (pusher.isEncryptedChannel(channel) && pusher.config.encryptionMasterKey !== undefined) {
```

Comment:
IMO this should result in an error being returned.

It failed to produce a "socket signature" because the encryptionMasterKey was not set.

Reply:
This one is a bit subtle.

If the client requests an auth for an encrypted channel, but I don't have an encryption master key, should I still give the client what I can, i.e. an auth token but not the per-channel secret?

Here I've opted to just give the client what I can. Alternatively, the logic could be:

```
  if (util.isEncryptedChannel(channel)) {
    if (pusher.config.encryptionMasterKey === undefined) {
      throw new Error("Cannot generate shared_secret because I don't have the master key");
    }
    result.shared_secret = Buffer(pusher.channelSharedSecret(channel)).toString('base64');
  }
```

Is this what you're suggesting @WillSewell ?

Reply:
Yes that is what I'm suggesting.

I do see your point, but I think in practice this case is most likely a misconfiguration by our users (forgetting to set the encryption master key), and it would be clearer what is wrong if it fails early.

A middle ground approach would be to log a warning if the encryption master key is not set. WDYT?

Reply:
Resolving because @jameshfisher opted for the code snippet in his comment.